### PR TITLE
Fix: Apps are restarted/restaged when are not modified

### PR DESCRIPTION
**Pull request by boyan-velinov**
Jira:LMCROSSITXSADEPLOY-622

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java [hunk 1]
```diff
@@ -233,4 +234,9 @@ private void bindService(CloudFoundryOperations client, String appName, String s
         return (bindingParameters == null) ? null : bindingParameters.get(serviceName);
     }
 
+    protected static Map<String, Object> getParametersForBinding(CloudServiceBinding cloudServiceBinding) {
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java]: Could this be renamed to getBindingParametersOrDefault? or something similar?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStep.java [hunk 2]
```diff
@@ -193,8 +193,8 @@ private boolean updateServices(CloudApplicationExtended app, CloudApplication ex
                 bindService(context, client, app.getName(), serviceName, bindingParametersForCurrentService);
                 continue;
             }
-            Map<String, Object> existingBindingParameters = existingBindingForApplication.getBindingOptions();
-            if (!Objects.equals(existingBindingParameters, bindingParametersForCurrentService)) {
+            Map<String, Object> existingBindingParameters = getParametersForBinding(existingBindingForApplication);//existingBindingParameters is empty linkedhashmap for CF / null for XS
+            if (!Objects.equals(existingBindingParameters, bindingParametersForCurrentService)) {  // bindingParametersForCurrentService is null
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStep.java]: The comments here are not needed?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java [hunk 3]
```diff
@@ -233,4 +234,9 @@ private void bindService(CloudFoundryOperations client, String appName, String s
         return (bindingParameters == null) ? null : bindingParameters.get(serviceName);
     }
 
+    protected static Map<String, Object> getParametersForBinding(CloudServiceBinding cloudServiceBinding) {
+        Map<String, Object> bindingParameters = cloudServiceBinding.getBindingOptions();
+        return (MapUtils.isEmpty(bindingParameters)) ? null : bindingParameters;
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java]: Why not use our method CommonUtil::isNullOrEmpty :) ?

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStepTest.java [hunk 4]
```diff
@@ -199,6 +245,24 @@ private void prepareDiscontinuedServices() {
         notRequiredServices.addAll(discontinuedServices);
     }
 
+    private boolean listContainsElement(List<String> list, String element) {
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStepTest.java]: Why not using List::contains method :D ?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java (hunk 3 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java]: Done

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStepTest.java (hunk 4 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStepTest.java]: Done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStep.java (hunk 2 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateAppStep.java]: Done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java (hunk 1 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java]: Done

**Pull request merged**